index.js

```diff
@@ -551,6 +552,19 @@ Browserify.prototype._createDeps = function (opts) {
                 return through();
             }
         }
+
+        if (opts.commondir === false && opts.builtins === false) {
+          opts.insertGlobalVars = xtend({
+            __dirname: function(file, basedir) {
+              var dir = path.dirname(path.relative(basedir, file));
+              return 'require("path").join(__dirname,"' + dir.split(path.sep).join('","') + '")';
+            },
+            __filename: function(file) {
+              var filename = path.relative(basedir, file);
```

Comment:
Looks like there is another `basedir` variable defined above (https://github.com/substack/node-browserify/blob/master/index.js#L428):

```js
var basedir = defined(opts.basedir, process.cwd());
```

And that happens to have the right one as well (its basically what gets passed to the `__filename` function).

I'll see if I can somehow break it and test it.